slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/SlashingProtection.java

```diff
@@ -34,7 +34,7 @@ boolean maySignBlock(
 
   void registerValidators(List<Bytes> validators);
 
-  boolean isRegisteredValidator(Bytes publicKey);
+  boolean slashingProtectionDataExistsFor(Bytes publicKey);
```

Comment:
nit: hasSlashingProtectionData would make clearer we are returning a boolean

Reply:
@jframe I've gone for hasSlashingProtectionDataFor as a compromise. 

Agree that hasX fits in with convention, however I also wanted to make clear that it was the passed in public key that has the data. This was to distinguish from the optional slashingProtection object in the expression...
```
final boolean slashingProtectionDataExistsForPubKey =
              slashingProtection
                  .map(sp -> sp.hasSlashingProtectionDataFor(Bytes.fromHexString(pubkey)))
                  .orElse(false);
```

Overall, this expression is saying "we have slashingProtection enabled and have slashing protection data for this key".

For the same reason, I have left the tightly scoped boolean `slashingProtectionDataExistsForPubKey` unchanged.